noaa_metrics/aggregate_logs.py

```diff
@@ -16,59 +20,278 @@
 )
 
 
+def get_available_memory_gb():
+    """Get available system memory in GB."""
+    return psutil.virtual_memory().available / (1024**3)
+
+
+def get_file_size_mb(filepath: Path) -> float:
+    """Get file size in MB."""
+    try:
+        return os.path.getsize(filepath) / (1024 * 1024)
+    except:
+        return 0
+
+
+def read_json_file_safe(filepath: Path) -> pd.DataFrame:
+    """Safely read a JSON file and return DataFrame."""
+    try:
+        if os.path.getsize(filepath) > 2:
+            return pd.read_json(filepath)
+        else:
+            return pd.DataFrame()
+    except Exception as e:
+        print(f"Error reading {filepath}: {e}")
+        return pd.DataFrame()
+
+
+def create_dataframe_streaming_fallback(
+    json_dir: Path, start_date: dt.date, end_date: dt.date
+) -> pd.DataFrame:
+    """
+    Emergency fallback: ultra-conservative streaming approach.
+    Processes one file at a time with aggressive memory management.
+    """
+    dates = pd.date_range(start_date, end_date, freq="d").tolist()
+    filepaths = [
+        json_dir / f"noaa-metrics-{date:%Y-%m-%d}.json" for date in dates
+    ]
+    
+    print("Using emergency streaming mode...")
+    
+    # Start with empty DataFrame
+    result_df = pd.DataFrame()
+    total_rows = 0
+    files_processed = 0
+    
+    for filepath in filepaths:
+        if filepath.is_file() and os.path.getsize(filepath) > 2:
+            size_mb = get_file_size_mb(filepath)
+            available_gb = get_available_memory_gb()
+            
+            print(f"  {filepath.name} ({size_mb:.1f}MB, {available_gb:.1f}GB available)")
+            
+            # Skip files that are too large for current memory
+            if size_mb > available_gb * 300:  # Conservative threshold
```

Comment:
Maybe define 300 as a constant.
And then in the print statement, include the size_mb and the 300*available_gb to give an indication of the "scale" of the issue that occurred here.